lib/assert.js

```diff
@@ -87,8 +87,18 @@ function wrapAssertions(callbacks) {
 			if (Object.is(actual, expected)) {
 				pass(this);
 			} else {
-				const actualDescriptor = concordance.describe(actual, concordanceOptions);
-				const expectedDescriptor = concordance.describe(expected, concordanceOptions);
+				const result = concordance.compare(actual, expected, concordanceOptions);
+				if (result.pass) {
+					return fail(this, new AssertionError({
+						assertion: 'is',
+						message,
+						raw: {actual, expected},
+						values: [formatWithLabel(`Objects are identical, but it's not the same object:`, actual)]
```

Comment:
This should use a regular string, not a template string.

As for the label itself, how about:

> Values are deeply equal to each other, but they're not the same:

This references both `t.deepEqual` and the ["Assert `value` is the same as `expected`" explanation](https://github.com/avajs/ava/tree/3b81e2ca13db91e558234f55badc312209677d82#isvalue-expected-message) of `t.is()`.

Reply:
Label sounds good for me!

I used it because of "it's". Linting style doesn't allow doublequotes.
So I changed your label a bit for this reason (they're ->> they are) and used regular string.

I also moved assignments up, so we don't run .describe second time in deeply equal case.)

Reply:
> I also moved assignments up, so we don't run .describe second time in deeply equal case.)

Nice!

> I used it because of "it's". Linting style doesn't allow doublequotes.
> So I changed your label a bit for this reason (they're ->> they are) and used regular string.

I don't know if we're consistent in this regard, but `’` is a good apostrophe to use. Or else the single quote can be escaped. Anyway, the copy is fine now.